geomesa-fs/geomesa-fs-storage/geomesa-fs-storage-parquet/src/main/scala/org/locationtech/geomesa/fs/storage/parquet/io/SimpleFeatureWriteSupport.scala

```diff
@@ -24,26 +26,49 @@ import java.util.{Date, UUID}
 
 class SimpleFeatureWriteSupport extends WriteSupport[SimpleFeature] {
 
+  private class SimpleObserver extends MetadataObserver {
+    override protected def onClose(bounds: Envelope, count: Long): Unit = {}
+  }
+
+  private val observer = new SimpleObserver
+
   private var writer: SimpleFeatureWriteSupport.SimpleFeatureWriter = _
   private var consumer: RecordConsumer = _
+  private var schema: SimpleFeatureParquetSchema = _
 
   override val getName: String = "SimpleFeatureWriteSupport"
 
   // called once
   override def init(conf: Configuration): WriteContext = {
-    val schema = SimpleFeatureParquetSchema.write(conf).getOrElse {
+    schema = SimpleFeatureParquetSchema.write(conf).getOrElse {
       throw new IllegalArgumentException("Could not extract SimpleFeatureType from write context")
     }
     this.writer = SimpleFeatureWriteSupport.SimpleFeatureWriter(schema.sft)
 
     new WriteContext(schema.schema, schema.metadata)
   }
 
+  // called once at the end after all SimpleFeatures are written
+  override def finalizeWrite(): FinalizedWriteContext = {
+    // Get the bounding box that spans all the SimpleFeatures' geometries
+    val envelope = observer match {
+      case obj: BoundsObserver => Some(obj.getBoundingBox)
```

Comment:
we shouldn't need a cast here?

Reply:
i think after looking at the geoparquet json that we actually need to track the bounds for each geometry separately, when there are multiple geometries in the schema.